src/Gateway/PayplugResponse.php

```diff
@@ -130,8 +133,6 @@ public function process_payment($resource, $is_payment_with_token = false, $sour
 				\do_action('payplug_gateway_payment_response_processed', $order_id, $resource);
 				PayplugWoocommerceHelper::set_flag_ipn_order($order, $metadata, false);
 				PayplugGateway::log(sprintf('Order #%s : '. $this->gateway_name($gateway_id) .' payment IPN %s processing completed successfully.', $order_id, $resource->id));
-			} elseif (( $resource->failure == null ) && ( $resource->payment_method['is_pending'] == true )) {
-				$this->handle_pending_oney( $order, $resource );
```

Comment:
shouldn't we reduce the stock if the `$resource` isn't a failure?

Reply:
reducing the stock would be automatically done if the order goes from Pending to In progress, there we are only handling the specific oney pending case